buildenv/jenkins/common/variables-functions

```diff
@@ -244,7 +244,12 @@ def set_build_variables() {
 def set_test_variables() {
     set_release()
     set_jdk_folder()
-    JAVA_VERSION = "SE" + "${SDK_VERSION}" + "0"
+    
+    if ( SPEC ==~ /.*valhalla_nestmates/) {
+        JAVA_VERSION = "Valhalla"
+    } else {
+        JAVA_VERSION = "SE" + "${SDK_VERSION}" + "0"
+    }
```

Comment:
Do you need to set this variable here (change this file at all?), would you not just set this variable where you set valhalla_nestmates specs?

Reply:
I think this is the correct spot as this is the only spot we set JAVA_VERSION (which is only used for test purposes). If we set it in the top level PR file then we would have to set it in every top level file we end up adding.

Reply:
Right, so stick with 1.  

I'd like to see a pass of simplification on the pipeline scripts separate from this PR.

The set_test_variables() name is confusing as the method only sets JAVA_VERSION (RELEASE and SDK_DIR are not directly used by testing, must be mapped to the variables that tests expect... somewhere else).

I had an expectation to set the few environment variables needed for test in one spot, rather than scattered around the different pipeline functions.   In that way, the pipeline code maps more closely to the standalone run on command line story, making it more readable/understandable/maintainable by developers who have run testing locally. We should have the goal of less levels of indirection.

I would suggest creating a naming standard for specs that gets followed, so this change doesn't become a giant if else, else else statement as it gets scaled up.  You could change the regex to match on any spec with valhalla in it to set version Valhalla, and not have to keep growing the if statement.

Reply:
I'll change the regex to check that it matches any spec whose name contains 'valhalla' instead of any check which ends in `valhalla_nestmates'

It looks like the other test variables are set within `buildenv/jenkins/common/test` and that the value of `JAVA_BIN` has to be set there. This is because `set_job_variables()` is called on the master machine and the `JAVA_BIN` value refers to a location on the node machine. The `JAVA_BIN` location is on the node machine because the tests run on the node. Since the `JAVA_BIN` is set as
> JAVA_BIN = "${WORKSPACE}/${BUILD_RELEASE_IMAGES}${JDK_FOLDER}${JRE_FOLDER}/bin"

and the value of `WORKSPACE` is different between the two machines, it can't be set while running `set_job_variables` on the master one.

I could try copying/moving the test variable definitions from the `variables-functions` script so that they run on the node machine and can be set together. However, this will cause some duplication since some variables also need to be set on the master machine

Reply:
So within this PR, I will add another commit with a comment in `set_test_variables()` which indicates/clarifies that `JAVA_BIN` is set elsewhere and then open an issue for further discussion?